JustTrack/JEEventsGenerator/main.swift

```diff
@@ -153,70 +153,77 @@ func generateStructs(_ events: [String : AnyObject]) throws -> NSString {
     let resultString: NSMutableString = NSMutableString(string: structListTemplateString)
     var structsArray: [String] = Array()
     
-    for eventName: String in events.keys {
-        
-        let eventDic: [String : AnyObject]? = events[eventName] as? [String : AnyObject]
-        
-        var structString = structTemplate
-        structString = replacePlaceholder(structString, placeholder: "<*!\(JETemplatePlaceholder.eventName.rawValue)*>", value: eventName) //<*!event_name*> = Example
-        structString = replacePlaceholder(structString, placeholder: "<*\(JETemplatePlaceholder.eventName.rawValue)*>", value: eventName) //<*event_name*> = example
-        
-        
-        let keys: [String] = eventDic![JEPlistKey.payload.rawValue] as! [String]
-        //<*event_keyValueChain*> = kKey1 : key1, kKey2 : key2
-        let eventKeyValueChain: String = generateEventKeyValueChain(keys)
-        structString = replacePlaceholder(structString, placeholder: "<*\(JETemplatePlaceholder.keyValueChain.rawValue)*>", value: eventKeyValueChain)
-        
-        //<*event_cs_trackers_str*> = "console", "GA"
-        let eventCsTrackers: String = generateEventCsTrackers(eventDic![JEPlistKey.trackrs.rawValue] as! [String])
-        structString = replacePlaceholder(structString, placeholder: "<*\(JETemplatePlaceholder.eventTrackers.rawValue)*>", value: eventCsTrackers)
+    for eventName in events.keys {
         
-        /*
-         <*event_keysNames*> =
-         private let kKey1 = "key1"
-         private let kKey2 = "key2"
-         */
-        let eventKeysNames: String = try generateEventKeysNames(keys)
-        structString = replacePlaceholder(structString, placeholder: "<*\(JETemplatePlaceholder.keysNames.rawValue)*>", value: eventKeysNames)
-        
-        /*
-         <*event_keysVars*> =
-         let key1 : String
-         let key2 : String
-         */
-        let eventKeysVars: String = try generateEventKeysVars(keys)
-        structString = replacePlaceholder(structString, placeholder: "<*\(JETemplatePlaceholder.keysVars.rawValue)*>", value: eventKeysVars)
-        
-        /*
-         <*event_init*> =
-         init(<*event_init_params*>) {
+        if let eventDic = events[eventName] as? [String : AnyObject] {
```

Comment:
if the eventName exist then the dictionary must exist, even if empty.
Could you please remove the if and revert to the previous implementation?
This will even make easier to review the PR, highlighting only the actual changes and not the entire method.